File: frontend/src/constants/ResourceStatus.ts
export const ResourceStatus = ["AVAILABLE", "BOOKED", "MAINTENANCE", "OFFLINE"] as const;
export type ResourceStatus = (typeof ResourceStatus)[number];
export const ResourceStatusText: Record<ResourceStatus, string> = {
  AVAILABLE: "可用",
  BOOKED: "已占用",
  MAINTENANCE: "维护中",
  OFFLINE: "停用"
};

File: backend/src/constants/ResourceStatus.go
package constants

var ResourceStatus = []string{"AVAILABLE", "BOOKED", "MAINTENANCE", "OFFLINE"}

const (
	ResourceStatusAvailable   = "AVAILABLE"
	ResourceStatusBooked      = "BOOKED"
	ResourceStatusMaintenance = "MAINTENANCE"
	ResourceStatusOffline     = "OFFLINE"
)

// UsableForRebook marks resource statuses that may receive a swapped booking.
var UsableForRebook = []string{ResourceStatusAvailable}
